File: Day 19 Interfaces/Day 19 Interfaces.py
class AdvancedArithmetic(object):
    def divisorSum(n):
        raise NotImplementedError

class Calculator(AdvancedArithmetic):
    def divisorSum(self, n):
        pass
class Calculator(AdvancedArithmetic):
    def divisorSum(self, n):
        add = [i for i in range(1, n+1) if n % i == 0]
        return sum(add)


n = int(input())
my_calculator = Calculator()
s = my_calculator.divisorSum(n)
print("I implemented: " + type(my_calculator).__bases__[0].__name__)
print(s)

File: Day 19 Interfaces/Day 19 Interfaces.cs
using System;
public interface AdvancedArithmetic{
    int divisorSum(int n);
}

public class Calculator : AdvancedArithmetic
{
    int sumOfDivisiors = 0;
    public int divisorSum(int n)
    {
        for(int i = 1; i <= n; i++)
        {
            if(n % i == 0)
            {
                sumOfDivisiors += i;
            }
        }
        return sumOfDivisiors;
    }
}

class Solution{
    static void Main(string[] args){
        int n = Int32.Parse(Console.ReadLine());
      	AdvancedArithmetic myCalculator = new Calculator();
        int sum = myCalculator.divisorSum(n);
        Console.WriteLine("I implemented: AdvancedArithmetic\n" + sum); 
    }
}